File: FE/WEB/app/models/user-profile.js
import Model, { attr, belongsTo } from '@ember-data/model';

export default class UserProfileModel extends Model {
  @belongsTo('user', { async: true, inverse: 'userProfile' }) user;

  @attr('string', { defaultValue: '' }) userId;
  @attr('string', { defaultValue: '' }) contactEmail;
  @attr('string', { defaultValue: '' }) firstName;
  @attr('string', { defaultValue: '' }) lastName;
  @attr('string', { defaultValue: '' }) nickName;
  @attr('string', { defaultValue: '' }) fixedPhone;
  @attr('string', { defaultValue: '' }) mobilePhone;
  @attr('string', { defaultValue: '' }) sex;
  @attr('string', { defaultValue: '' }) taxId;

  @attr('date') birthDate;
  @attr('string', { defaultValue: '' }) birthCity;
  @attr('string', { defaultValue: '' }) birthProvince;
  @attr('string', { defaultValue: '' }) birthZIP;
  @attr('string', { defaultValue: '' }) birthState;

  @attr('string', { defaultValue: '' }) residenceCity;
  @attr('string', { defaultValue: '' }) residenceProvince;
  @attr('string', { defaultValue: '' }) residenceZIP;
  @attr('string', { defaultValue: '' }) residenceState;
  @attr('string', { defaultValue: '' }) residenceAddress;
  @attr('string', { defaultValue: '' }) residenceHouseNumber;

  @attr('string', { defaultValue: '' }) occupation;
  @attr('string', { defaultValue: '' }) description;

  @attr('string', { defaultValue: '' }) profileImageId;

  @attr('string', { defaultValue: '' }) profileFreeFieldString1;
  @attr('string', { defaultValue: '' }) profileFreeFieldString2;
  @attr('string', { defaultValue: '' }) profileFreeFieldString3;
  @attr('date') profileFreeFieldDateTime;
  @attr('number', { defaultValue: 0 }) profileFreeFieldInt1;
  @attr('number', { defaultValue: 0 }) profileFreeFieldInt2;
  @attr('boolean', { defaultValue: false }) profileFreeFieldBoolean;

  @attr('string', { defaultValue: '' }) userLang;

  @attr('string', { defaultValue: 'limited' }) cookieAccepted;
  @attr('boolean', { defaultValue: false }) termsAccepted;
  @attr('date') termsAcceptanceDate;
  @attr('date') registrationDate;
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/UserProfile.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Table("UserProfile")]
    public class UserProfile : Identifiable<string>
    {
        [Attr] public string UserId { get; set; }
        [Attr] public string FirstName { get; set; }
        [Attr] public string LastName { get; set; }
        [Attr] public string NickName { get; set; }
        [Attr] public string FixedPhone { get; set; }
        [Attr] public string MobilePhone { get; set; }
        [Attr] public string Sex { get; set; }
        [Attr] public string TaxId { get; set; }
        [Attr] public DateTime? BirthDate { get; set; }
        [Attr] public string BirthCity { get; set; }
        [Attr] public string BirthProvince { get; set; }
        [Attr] public string BirthZIP { get; set; }
        [Attr] public string BirthState { get; set; }
        [Attr] public string ResidenceCity { get; set; }
        [Attr] public string ResidenceProvince { get; set; }
        [Attr] public string ResidenceZIP { get; set; }
        [Attr] public string ResidenceState { get; set; }
        [Attr] public string ResidenceAddress { get; set; }
        [Attr] public string ResidenceHouseNumber { get; set; }
        [Attr] public string Occupation { get; set; }
        [Attr] public string Description { get; set; }
        [Attr] public string ContactEmail { get; set; }
        [Attr] public string ProfileImageId { get; set; }
        [Attr] public string ProfileFreeFieldString1 { get; set; }
        [Attr] public string ProfileFreeFieldString2 { get; set; }
        [Attr] public string ProfileFreeFieldString3 { get; set; }
        [Attr] public string AppleRefreshToken { get; set; }
        [Attr] public string GoogleRefreshToken { get; set; }
        [Attr] public string FacebookRefreshToken { get; set; }
        [Attr] public string TwitterRefreshToken { get; set; }
        [Attr] public string UserLang { get; set; }
        [Attr] public int? ProfileFreeFieldInt1 { get; set; }
        [Attr] public int? ProfileFreeFieldInt2 { get; set; }
        [Attr] public DateTime? ProfileFreeFieldDateTime { get; set; }
        [Attr] public bool? ProfileFreeFieldBoolean { get; set; }

        [Attr] public string cookieAccepted { get; set; }
        [Attr] public bool? termsAccepted { get; set; }
        [Attr] public DateTime? termsAcceptanceDate { get; set; }
        [Attr] public DateTime? registrationDate { get; set; }
        [Attr] public string birthRegion { get; set; }
        [Attr] public string residenceRegion { get; set; }

        [HasOne]
        public virtual User User { get; set; }
    }
}
